stdlib/format.mli

```diff
@@ -1472,6 +1472,82 @@ val kasprintf : (string -> 'a) -> ('b, formatter, unit, 'a) format4 -> 'b
   @since 4.03
 *)
 
+(** Formatted Pretty-Printing with heterogeneous list as arguments.
+
+  The following functions behave the same as their non-'l' counter-parts, but
+  receive their arguments as a heterogeneous list.
+
+  The heterogeneous list serves as a syntactically-delimited
+  collection of arguments, eliminating the need for continuation variants.
+  This approach also improves the clarity of type error messages,
+  especially when there is a mismatch between the format string and the
+  number of arguments provided.
+
+  The heterogeneous list form of [dprintf] can be easily derived
+  from [lfprintf], as [dprintf] is the same as [fprintf] with its formatter
+  as the last argument.
+
+  For example:
+{[
+  let ldprintf fmt args ppf = Format.lfprintf ppf fmt args;;
+  let t = ldprintf "%i@ %s@ %.02f" [ 42; "ocaml"; 3.14 ];;
+  ...
+  Format.lprintf "@[<v>%t@]" [ t ]
+]}
+*)
+
+module Arg_list : sig
+  type ('a, 'r) t =
+    | [] : ('r, 'r) t
+    | (::) : 'a * ('b, 'r) t -> ('a -> 'b, 'r) t
+
+  val apply : 'a -> ('a, 'r) t -> 'r
+
+  val ( @ ) : ('a, 'r1) t -> ('r1, 'r2) t -> ('a, 'r2) t
+end
+(** Contains the type (and append function) of the heterogeneous list used
+    as arguments for the printf-like functions that accept it.
```

Comment:
I believe that describing the intents of the module provide a better way to introduce its contents:

```
The [Arg_list] module defines a heterogeneous list type, that can be used as argument of [printf]-like functions.
```
if you want to mention the append and apply function, you could add
```
and various related functions.
```
but this is the case for most modules, thus it is probably fine to leave this point out.

Reply:
Agreed! The newest commit has the updated the module description.